src/core/component.js

```diff
@@ -390,7 +398,7 @@ module.exports.registerComponent = function (name, definition) {
  * @param {boolean} silent - Suppress warning messages.
  * @return {object} The component data
  */
-function buildData (el, name, attrName, schema, elData, silent) {
+function buildData (el, name, attrName, schema, elData, oldElData, silent, skipTypeChecking) {
```

Comment:
We should just plop this back into the component definition, too many arguments.